chrome/content/zotero/preferences/preferences_sync.jsx

```diff
@@ -182,6 +182,19 @@ Zotero_Preferences.Sync = {
 			Zotero.Sync.Runner.deleteAPIKey();
 			return;
 		}
+		
+		// It shouldn't be possible for a sync to be in progress if the user wasn't logged in,
+		// but check to be sure
+		if (!Zotero.Sync.Runner.syncInProgress) {
+			// Clear any displayed sync errors
+			Zotero.Sync.Runner.updateIcons([]);
+		}
+		window.addEventListener('beforeunload', () => {
+			if (!Zotero.Sync.Runner.syncInProgress) {
+				Zotero.Sync.Runner.sync();
```

Comment:
This will start a sync even if auto-sync is disabled. Could check `Zotero.Prefs.get('sync.autoSync')`, or could just use `Zotero.Sync.Runner.setSyncTimeout(1)` instead, which does the check. Maybe a little less jarring to wait a second anyway?